### PR TITLE
PLG-352: As UX designer, I want the PIM navigation to be consistent with the DSM components

### PR DESCRIPTION
**Pull request by rouliane**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/menu_new.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    });
+
+    const entries: NavigationEntry[] = navigationEntriesExtensions.map((extension: EntryView) => {
+      const {title, isLandingSectionPage} = extension.config;
+      return {
+        code: extension.code,
+        title: title,
+        disabled: false,
+        route: this.findEntryRoute(extension),
+        icon: React.createElement(CardIcon),
```
Comment (rouliane) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/menu_new.ts:123]: Custom icon + size 24 x 24px

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/requirejs.yml [hunk 2]
```diff
@@ -920,16 +920,16 @@ config:
     pim/form/user/login-details:                         pimui/js/form/user/login-details
     pim/form/common/exclusif-boolean: pimui/js/form/common/exclusif-boolean
 
-    pim/menu/menu:                                       pimui/js/menu/menu
-    pim/menu/tab:                                        pimui/js/menu/tab
-    pim/menu/item:                                       pimui/js/menu/item
+    pim/menu/menu:                                       pimui/js/menu/menu_new/menu_new.ts
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/config/requirejs.yml:923]: we should find a better path and file names

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  IconButton,
+  IconProps,
+  MainNavigationItem,
+  MoreVerticalIcon,
+  SubNavigationItem,
+  SubNavigationPanel,
+  useBooleanState,
+} from 'akeneo-design-system';
+
+type NavigationEntry = {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx:15]: Types could be defined in a proper file

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        </HelpContainer>
+      </MainNavContainer>
+      {activeNavigationEntry &&
+      (!activeNavigationEntry.isLandingSectionPage || activeSubEntryCode) &&
+      activeSubNavigation &&
+      activeSubNavigation.sections.length > 0 && (
+        <SubNavContainer>
+          <SubNavigationPanel isOpen={isSubNavigationOpened} open={openSubNavigation} close={closeSubNavigation}>
+            {!isSubNavigationOpened &&
+              <Dropdown>
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx:119]: In order to make the code more readable, we could move the SubNavigation Dropdown in a dedicated component

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+                    title=''
+                    icon={<MoreVerticalIcon />}
+                    ghost="borderless"
+                    onClick={openMenu}
+                    className="dropdown-button"
+                  />
+                  {isMenuOpen &&
+                    <Dropdown.Overlay onClose={closeMenu}>
+                      <Dropdown.Header>
+                        {activeSubNavigation.title && <Dropdown.Title>{translate(activeSubNavigation.title)}</Dropdown.Title>}
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/PimNavigation.tsx:131]: Do we need to display the dropdown header if there is no title property defined in the activeSubNavigation?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+import {ViewOptions} from 'backbone';
+
+const BaseColumn = require('pim/form/common/column');
+const router = require('pim/router');
+const mediator = require('oro/mediator');
+
+/**
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:9]: Do we need to keep the copyright in Frontend files? 
If we do, could you update the author and dates, please?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/menu_new.ts [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+      this.el
+    );
+
+    return this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  renderExtension(extension: any) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/menu_new.ts:95]: Does the override of the `renderExtension` method still used?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/navigation-block_new.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+      });
+    });
+
+    return super.configure();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  render() {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/navigation-block_new.ts:70]: The render of the block is not the responsibility of the view anymore, we should remove the `render` method here

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/navigation-block_new.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+ * Base extension for navigation blocks
+ * A navigation block is composed of a title and a list of items, displayed in the columns.
+ *
+ * @copyright 2021Z Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class NavigationBlock extends BaseForm {
+  template;
+
+  constructor(options?: ViewOptions<any>) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/navigation-block_new.ts:28]: We do not need of the template and the classe name anymore

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/tab_new.ts [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+  configure() {
+    this.listenTo(this.getRoot(), 'pim_menu:register_item', this.registerItem);
+
+    return super.configure();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  render() {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/tab_new.ts:82]: The render of the block is not the responsibility of the view anymore, we should remove the render method here

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/tab_new.ts [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Tab extends BaseForm {
+  template;
+  active;
+  items: any[];
+  sections: any[];
+
+  constructor(options?: ViewOptions<any>) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/tab_new.ts:37]: We do not need the `events`,  `className` and `template` properties anymore

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/item_new.ts [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+      routeParams: this.getRouteParams(),
+    });
+
+    return super.configure();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  render() {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/item_new.ts:77]: The render of the block is not the responsibility of the view anymore, we should remove the render method here

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/item_new.ts [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+ *
+ * @author    Julien Sanchez <julien@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Item extends BaseForm {
+  template;
+  active;
+
+  constructor(options?: ViewOptions<any>) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/item_new.ts:32]: We do not need of the `template` and `events` properties anymore

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+  configure() {
+    this.onExtensions('pim_menu:column:register_navigation_section', this.registerNavigationSection);
+
+    return super.configure();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  render() {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:49]: The render of the Column is not the responsibility of the Backbone View anymore, we should remove the render method here

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+    } else {
+      this.active = event.extension === this.getTab();
+    }
+    this.isVisible = true;
+
+    this.render();
+  }
+
+  hide(menuIdentifier: string) {
+    if (this.code === menuIdentifier) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:76]: We should delegate this event to the parent module (`Menu`) and let it handles the update of the state and the re-render of the `PimNavigation` component

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  /**
+   * Highlight or un-highlight tab
+   *
+   * @param {Event} event
+   * @param {string} event.extension The extension code to highlight
+   * @param {string} event.columnExtension The extension code of the column to activate
+   */
+  highlight(event: {extension: string; columnExtension?: string}) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:64]: We should delegate this event to the parent module (`Menu`) and let it handles the update of the state and the re-render of the `PimNavigation` component

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+  getTab() {
+    return this.config.tab;
+  }
+
+  /**
+   * The DOM element contains a `data-tab` attribute for compatibility with tab Bootstram tabs.
+   *
+   * {@inheritdoc}
+   */
+  redirect(event: any) {
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:97]: We should delegate this event to the parent module (`Menu`) and let it handles the update of the state and the re-render of the `PimNavigation` component

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts (hunk 15 above)
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:76]: We should delegate this event to the parent module (`Menu`) and let it handles the update of the state and the re-render of the `PimNavigation` component

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts (hunk 17 above)
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/menu/menu_new/column_new.ts:97]: Do we still need to handle the redirection here or do we delegate the event to the parent `Menu` module?

File: front-packages/shared/src/components/Navigation/PimNavigation.unit.tsx [hunk 18]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {mockedDependencies, renderWithProviders} from '../../tests';
+import {fireEvent, screen} from '@testing-library/react';
+import {PimNavigation} from './PimNavigation';
+import {aMainNavigation} from './navigationHelper';
+
+jest.mock('../PimView');
+
```
Comment (olivierpontier) [inline front-packages/shared/src/components/Navigation/PimNavigation.unit.tsx:8]: In order to ease the reading of the cli output, could you wrap the tests in a describe block please? 
```
describe('PimNavigation', () => {
  // ... test blocks here
});
```

**Comment by olivierpontier** [comment]
In order to ease the reading of the cli output, could you wrap the tests in a describe block please? 
```
describe('Module name', () => {
  // ... test blocks here
});
```

File: front-packages/shared/src/components/Navigation/SubNavigationDropdown.unit.tsx [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+  renderWithProviders(
+    <SubNavigationDropdown title='Sub navigation title' entries={subNavigationEntries}/>
+  );
+
+  fireEvent.click(screen.getByTestId('openSubNavigationDropdownButton'));
+  expect(screen.getByText('Sub navigation title')).toBeInTheDocument();
+  expect(screen.getByText('Sub entry 1')).toBeInTheDocument();
+  expect(screen.getByText('Sub entry 2')).toBeInTheDocument();
+  expect(screen.getByText('Sub entry 3')).toBeInTheDocument();
+  expect(screen.getByText('Sub entry 3')).toBeInTheDocument();
```
Comment (rouliane) [inline front-packages/shared/src/components/Navigation/SubNavigationDropdown.unit.tsx:28]: duplicated line to remove

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/navigation-menu/menu.ts [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+    to?: string;
+    isLandingSectionPage?: boolean;
+    tab?: string;
+    icon: string;
+  };
+  items: SubEntry[];
+  sections: any[];
+};
+
+// @fixme Define what is an entry column
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/navigation-menu/menu.ts:36]: Hey @LaurentPetard @rouliane,
did we took account of this comment?

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/navigation-menu/menu.ts (hunk 20 above)
Comment (rouliane) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/navigation-menu/menu.ts:36]: I have no idea what this comment means...

Comment (LaurentPetard) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/navigation-menu/menu.ts:38]: I don't remember. I think it's a remaining comment of the POC

**Review by LaurentPetard — APPROVED** [review]

**Review by BryanS29-akeneo — APPROVED** [review]

**Pull request merged**